src/NodeServices/WorkspaceEvents.cs

```diff
@@ -87,7 +87,13 @@ internal static void OnWorkspaceSettingsChanged(bool enableLegacyPolyCurveBehavi
             var handler = WorkspaceEnableLegacyPolyCurveSettingChanged;
             if (handler != null)
             {
-                handler(new WorkspacesSettingsChangedEventArgs(enableLegacyPolyCurveBehavior));
+                try
+                {
+                    handler(new WorkspacesSettingsChangedEventArgs(enableLegacyPolyCurveBehavior));
+                }
+                catch (NullReferenceException)
```

Comment:
what is null in this case?

Reply:
is it hard to add a test for this? I would not have expected this to need a try catch.

Reply:
I think `mGeometryFactory` is null in this [line](https://git.autodesk.com/Dynamo/LibG/blob/49d9579b9548e668c860f9d9ad0a41d0a05d6d89/ProtoGeometry/Core/HostFactory.cs#L169) when ASM is not found.